### ISSUE TITLE
Agent monitoring server is binding to all interfaces instead of localhost

### ISSUE THREAD

**Issue by andrewkroh** [elastic/elastic-agent#2509]
For confirmed bugs, please report:
- Version: 8.6.2 (build: 913c02bea9b13dec4d5c5f3057b5b397344e3298 at 2023-02-13 16:52:36 +0000 UTC)
- Operating System: macOS 13.3.1 (22E261)
- Steps to Reproduce:
1. Deploy a Fleet managed Elastic Agent on macOS.
2. Observe the listening ports. (use netstat, Little Snitch, etc)

```json
[
  {
    "connections": [
      {
        "bytesDown": "0",
        "bytesUp": "0",
        "interface": "",
        "localAddress": "::",
        "localPort": "6791",
        "protocol": "TCP",
        "remoteAddress": "::",
        "remoteHostName": "::",
        "remotePort": "0",
        "state": "Listen"
      }
    ],
    "process": {
      "path": "/Library/Elastic/Agent/data/elastic-agent-913c02/elastic-agent.app/Contents/MacOS/elastic-agent",
      "pid": "837",
      "signature(s)": {
        "signatureStatus": 100013
      }
    }
  }
]
```

```
netstat -an | grep 6791
tcp46      0      0  *.6791                 *.*                    LISTEN
```

I expected to see something like this where it binding only to the loopback interface.

```
# MOCK DATA
tcp4       0      0  127.0.0.1.6791         *.*                    LISTEN
```

This may be a bug because the Agent code already appears to bind to localhost by default.

https://github.com/elastic/elastic-agent/blob/d7adf66bc4e462abed24b95ee48b0671b8a952ab/internal/pkg/core/monitoring/config/config.go#L56-L60

**Comment by WiegerElastic** [comment]
Wanted to share the output on my machine (MacOS 13.3.1) running Agent version 8.7.0.

<img width="640" alt="image" src="https://user-images.githubusercontent.com/97174155/232885118-18de52a1-37f8-47b6-9e65-b8748a48386f.png">

<img width="779" alt="image" src="https://user-images.githubusercontent.com/97174155/232887402-74e260af-8e3a-4a9c-ac4b-59bebb9beb0f.png">

**Comment by SpencerLN** [comment]
Are there any plans to prioritize fixing this? 

I noticed that `elastic-agent` seems to listen on all interfaces even when `agent.monitoring.http.enabled: false` is set:
```
agent:
  download:
    sourceURI: https://artifacts.elastic.co/downloads/
  features: null
  headers: null
  id: 3ab3a5fc-a6e7-4efd-8204-e745f6a63e5d
  logging:
    level: info
  monitoring:
    enabled: true
    http:
      buffer: null
      enabled: false
      host: ""
      port: 6791
```

```
% sudo lsof -PiTCP -sTCP:LISTEN +c0 | grep elastic-agent
elastic-agent              59884  root    9u  IPv6 0x68956d5be803a143      0t0  TCP *:6791 (LISTEN)
elastic-agent              59884  root   16u  IPv4 0x68956d6574fc23ab      0t0  TCP localhost:6789 (LISTEN)
elastic-agent              59884  root   17u  IPv4 0x68956d6574fde3ab      0t0  TCP localhost:6788 (LISTEN)
```

**Comment by pierrehilbert** [comment]
We are currently planning to fix this in 8.9.1 as we are considering that as a P2.
Did I misunderstand the criticality here?

**Comment by pierrehilbert** [comment]
I will postpone the fix here as it's still not a priority compare to some other bug to fix.

**Comment by cmacknz** [comment]
Bringing this into the current sprint since it makes sense to do this along with https://github.com/elastic/elastic-agent/issues/2734, see https://github.com/elastic/elastic-agent/issues/2734#issuecomment-1758493579

**Comment by cmacknz** [comment]
After some more investigation by Michal, we need to change the pre-configured policy in Kibana so that this won't be a breaking change in cloud. This is now blocked by https://github.com/elastic/kibana/issues/168629.

**Comment by michalpristas** [comment]
@cmacknz comment probably related to other issue. 
on the note of this issue.

anybody able to repro recently? i'm trying but all i see is 
```
tcp4       0      0  127.0.0.1.6791         *.*                    LISTEN
```
this should be good

**Comment by cmacknz** [comment]
> I think this comes into product decision whether we want to support listening on all interfaces when explicitly configured or not. My 2 cents would be towards No and guarding at the time of server creation for valid values.

Agree. We should not silently be listening on all interfaces, this is unexpected and may be causing agents to be accidentally exposed to the internet.
